### PR TITLE
Rewrite the component to Typescript + other improvements.

### PR DESCRIPTION
**Pull request by jansiegel**
This commit should fix most of the issues from the #64 epic:

- [x] `Dist` should have UMD, CommonJS and ES files. We can use Rollup as we did with React wrapper.
- [x] Add Issue template
- [x] Add Pull Request template
- [x] Add missing info to `package.json` #60, #61, #62 
- [x] Use `id` instead `root` and remove `this.` #18 **breaking change**
- [x] Make sure we can pass `id`, `style` and `class` attributes to HoT container
- [x] Add Typescript definition file or "rewrite" to Typescript #25
- [x] Add sourcemaps for `min.(js|css)` files (and `.js` file if we go the Typescript path)
- [x] Fix two remaining bugs #50 and #20
- [x] Check if workaround for  #65 is still needed

Additionally, `.table` component property was renamed to `.hotInstance` (**breaking change**).

### CODE REVIEW

**Comment by jansiegel** [comment]
41a1b8f introduces yet another **breaking change**:

The _'on-'_ prefixes were removed from hooks, so from now on, the wrapper configuration would be compatible to Handsontable config.

File: src/common/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+ * @private
+ */
+function preventInternalEditWatch(component) {
+  component.hotInstance.addHook('beforeChange', () => {
+    component.__internalEdit = true;
+  });
+}
+
+/**
+ * Generate an object containing all the available Handsontable properties and plugin hooks (with the `on`-prefixes added).
```
Comment (wojciechczerniak) [inline src/common/helpers.ts]: `'on'-prefixes` comment is no longer valid

File: src/common/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+ * Rewrite the settings object passed to the watchers to be a clean array/object prepared to use within Handsontable config.
+ *
+ * @param {*} observerSettings Watcher object containing the changed data.
+ * @returns {Object|Array}
+ */
+export function rewriteSettings(observerSettings): any[] | object {
+  let settings: any[] | object | null = null;
+  let type: { array?: boolean, object?: boolean } = {};
+
+  if (Object.prototype.toString.call(observerSettings).indexOf('Array') > -1) {
```
Comment (wojciechczerniak) [inline src/common/helpers.ts]: `Array.isArray()` is not enough?

File: src/common/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    settings = [];
+    type.array = true;
+
+  } else if (typeof observerSettings === 'object') {
```
Comment (wojciechczerniak) [inline src/common/helpers.ts]: `typeof null === 'object'` will be `true`. Does it suppose to include `observeSettings === null`?

File: src/common/HotTable.vue [hunk 4]
```diff
@@ -1,25 +1,41 @@
...
+    beforeDestroy: function () {
+      this.hotInstance.destroy();
+    }
   };
+
+  export default HotTable;
+  export {HotTable};
 </script>
 
 <style>
```
Comment (wojciechczerniak) [inline src/common/HotTable.vue]: Can we remove empty `<style></style>` ?

**Review by wojciechczerniak — DISMISSED** [review]

**Review by wojciechczerniak — APPROVED** [review]

**Pull request merged**